mne/inverse_sparse/_gamma_map.py

```diff
@@ -290,21 +292,25 @@ def gamma_map(evoked, forward, noise_cov, alpha, loose="auto", depth=0.8,
         in_pos = 0
         if len(X) < 3 * len(active_src):
             X_xyz = np.zeros((3 * len(active_src), X.shape[1]), dtype=X.dtype)
+            X_ori_ = np.zeros((3 * len(active_src), X.shape[1]), dtype=X.dtype)
             for ii in range(len(active_src)):
                 for jj in range(3):
                     if in_pos >= len(active_set):
                         break
                     if (active_set[in_pos] + jj) % 3 == 0:
                         X_xyz[3 * ii + jj] = X[in_pos]
+                        X_ori_[3 * ii + jj] = X_ori[in_pos]
```

Comment:
XXX: There is still a problem here, I have to check what is happening when we give `xyz_same_gamma=False`